File: calculate_average_grade/python_version/calculate_average_grade.py
from typing import List
from functools import reduce

def calculate_average_grade(grades: List[int]) -> float:
    if len(grades) == 0: return 0.0
    def sumList(accum, currVal): return accum + currVal
    # lambda accum, currVal: accum + currVal, grades
    return reduce(sumList, grades) / len(grades)

if __name__ == "__main__":
    print(calculate_average_grade([1, 2, 3, 4, 5])) # 3.0

File: calculate_average_grade/JS_version/calculateAverageGrade.js
/**
 * Calculates the average grade from an array of grades.
 *
 * @param {number[]} grades - An array of grades.
 * @returns {number} - The average grade.
 */
function calculateAverageGrade(grades) {
    return grades.length === 0 ?
        0 : grades.reduce((accum, curVal) => accum + curVal) / grades.length;
}


// --- jest issues ---
// https://stackoverflow.com/questions/35756479/does-jest-support-es6-import-export
// THIS DOESN'T WORK
//export default calculateAverageGrade
// BUT THIS DOES
module.exports = calculateAverageGrade
